apps/userdashboard/api.py

```diff
@@ -62,17 +58,23 @@ def get_permission_object(self):
         return self.project
 
     def get_queryset(self):
-        all_comments_project = helpers.get_all_comments_project(self.project)
-        num_reports = Count("reports", distinct=True)
-        return all_comments_project.annotate(num_reports=num_reports).annotate(
-            has_reports=ExpressionWrapper(
-                Q(num_reports__gt=0), output_field=BooleanField()
-            )
+        comments = helpers.get_all_comments_project(self.project)
+        comments = comments.annotate(
+            num_reports=Count("reports", distinct=True),
         )
```

Comment:
isn't the reposts here refering to the report table? and not the new ai-report?

Reply:
yes, good catch, it must be the sum of both. I'll make the correction.

Reply:
the helpers get_num_reports needs update too. and you can then use it here directly. also this test can be updated with the ai_report factory too.

Reply:
good catch. It looks like this function `apps.projects.helpers.py::get_num_reports` is not actually used anywhere. I think we can actually remove it.
In the `api.py` I need to annotate with the number of reports for a given comment This helper function counts the number of reports for a whole project (and returns and int).

Reply:
> also this test can be updated with the ai_report factory too

I just looked at `test_serializers.py`. Which test do you mean? I can't find any ai report in there.

Reply:
ok, resolved. we agreed on 1) modifiying the test `test_num_reports` in `test_serializers.py` and 2) the helpers function `get_num_reports` can be refactored, but since it is used nowhere we can also leave it as is or remove it altogether. I will look into refactoring it, but stop it it takes too long.